vendor/processors/stripe_processor.py

```diff
@@ -341,26 +352,46 @@ def build_price(self, offer, price):
         if 'stripe' not in offer.meta or 'product_id' not in offer.meta['stripe']:
             raise TypeError(f"Price cannot be created without a product_id on offer.meta['stripe'] field")
 
+        if isinstance(price, (int, float)):
+            currency = DEFAULT_CURRENCY
+            unit_amount = self.convert_decimal_to_integer(price)
+            is_price_obj = False
+        else:
+            currency = price.currency
+            unit_amount = self.convert_decimal_to_integer(price.cost)
+            is_price_obj = True
+
+
         price_data = {
             'product': offer.meta['stripe']['product_id'],
-            'currency': price.currency,
-            'unit_amount': self.convert_decimal_to_integer(price.cost),
-            'metadata': {'site': offer.site.domain, 'pk': price.pk}
+            'currency': currency,
+            'unit_amount': unit_amount,
+            'metadata': {'site': offer.site.domain}
         }
+
+        if is_price_obj:
+            price_data['metadata']['pk'] = price.pk
+        else:
+            price_data['metadata']['msrp'] = price
         
         if offer.terms < TermType.PERPETUAL:
             price_data['recurring'] = {
                 'interval': 'month' if offer.term_details['term_units'] == TermDetailUnits.MONTH else 'year',
                 'interval_count': offer.term_details['payment_occurrences'],
-                'usage_type': 'license'
+                'usage_type': 'licensed'
             }
         
         return price_data
     
     def build_coupon(self, offer, price):
+        if isinstance(price, (int, float)):
+            currency = DEFAULT_CURRENCY
+        else:
+            currency = price.currency
```

Comment:
This else will never happen in the sync if you always call `offer.current_price()`